zookeeper-server/src/main/java/org/apache/zookeeper/server/NIOServerCnxnFactory.java

```diff
@@ -820,7 +828,12 @@ private void addCnxn(NIOServerCnxn cnxn) throws IOException {
     }
 
     protected NIOServerCnxn createConnection(SocketChannel sock, SelectionKey sk, SelectorThread selectorThread) throws IOException {
-        return new NIOServerCnxn(zkServer, sock, sk, this, selectorThread);
+        if (zkServer != null) {
```

Comment:
given your fix above, is this check still useful ?

Reply:
@eolivelli 

here is not required, but can check again. So I think adding  it is well